Source/Utils/WarpXUtil.cpp

```diff
@@ -213,6 +213,77 @@ WarpXParser makeParser (std::string const& parse_function, std::vector<std::stri
     return parser;
 }
 
+void CheckGriddingForRZSpectral ()
+{
+#if (defined WARPX_DIM_RZ) && (defined WARPX_USE_PSATD)
+// When using the RZ spectral solver, the Hankel transform cannot be
+// divided among multiple blocks. Each block must extend over the
+// entire radial extent.
+// The grid can be divided up along z, but the number of blocks
+// must be >= the number of processors.
+
+    int max_level;
+    Vector<int> n_cell(AMREX_SPACEDIM, -1);
+
+    ParmParse pp_amr("amr");
+
+    pp_amr.get("max_level",max_level);
+    pp_amr.getarr("n_cell",n_cell,0,AMREX_SPACEDIM);
+
+    Vector<int> blocking_factor_x(max_level+1);
+    Vector<int> max_grid_size_x(max_level+1);
+
+    // Set the radial block size to be equal to the radial grid size.
+    blocking_factor_x[0] = n_cell[0];
+    max_grid_size_x[0] = n_cell[0];
+
+    for (int lev=1 ; lev <= max_level ; lev++) {
+        blocking_factor_x[lev] = blocking_factor_x[lev-1]*refRatio(lev-1);
+        max_grid_size_x[lev] = max_grid_size_x[lev-1]*refRatio(lev-1);
+    }
+
+    pp_amr.addarr("blocking_factor_x", blocking_factor_x);
+    pp_amr.addarr("max_grid_size_x", max_grid_size_x);
+
+    // Adjust the longitudinal block sizes, making sure that there are
+    // more blocks than processors.
+    int nprocs = ParallelDescriptor::NProcs();
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(n_cell[1] >= 2*nprocs,
```

Comment:
When bilinear filter is used, or when the shape factor order is >2, and depending on the staggering, I am afraid that 2 longitudinal cells would give wrong results. Could we set it to a larger value, say 8, to be on the safe side? And then, below, we could enforce it. 
```suggestion
    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(n_cell[1] >= 8*nprocs,
```

Reply:
Probably a good idea. But, is this checked anywhere, in case the user sets the blocking factor too small for the shape factor?

Reply:
No, and that's a good point, it should also be in the general sanity checks.

Reply:
I increased it to 8